cm2350/e200_intc.py

```diff
@@ -161,31 +161,13 @@ def handleException(self):
         with self.lock:
             newexc = self.pending.pop(0)
 
-        # If this was a debug exception and the external debugger is connected 
-        # it should have been handled through the external debugger by now, 
-        # don't change the PC
-        #
-
-        if isinstance(newexc, intc_exc.ResetException):
-            raise newexc
-
-        elif isinstance(newexc, intc_exc.DebugException) and \
-                self.emu.gdbstub.isClientConnected():
-            # Before halting re-evaluate if there are any pending interrupts. If 
-            # a stepi happens it may be done from the GDB server thread in which 
-            # case we don't want this event to appear to be an event that has to 
-            # be handled again.
-            self.hasInterrupt = self.pending and self.curlvl > self.pending[0].prio
-
-            self.emu._do_halt()
-
```

Comment:
Can you explain the change here?  Is this because the required handling is written into the Exception classes?

Reply:
The handling for the DebugException was redundant and was causing me extra lockups when I was trying to get break/continue/stepi working. I think this particular commit was an intermediate change where I removed the handling of ResetException and DebugExceptions but I did have to add them back in. the bigger change was that we don't halt in the `handleException` function, instead we want to raise the exception so that the standard `except DebugException` case catches it and then all debug events can be handled the same way.